code/modules/mob/inventory.dm

```diff
@@ -423,8 +419,15 @@
 	set hidden = TRUE
 
 	var/obj/item/I = get_active_held_item()
-	if (I)
-		I.equip_to_best_slot(src)
+	if(!I)
+		to_chat(src, "<span class='warning'>You are not holding anything to equip!</span>")
+		return
+	if (temporarilyRemoveItemFromInventory(I) && !QDELETED(I)) //Try remove it from hand slot, make sure that didn't destroy it.
+		if(I.equip_to_best_slot(src)) // Try to put in appriopriate slot
```

Comment:
```suggestion
		if(I.equip_to_best_slot(src)) // Try to put in appropriate slot
```

(though i think these comments are sort of useless anyway)

Reply:
Yeah, they're pretty redundant. I just hope that spelling these out will stop people from repeating the same inventory errors for the 100th time.